lib/dynamodb/dynamodb_request_queue.c

```diff
@@ -170,9 +147,10 @@ logreq(struct logging_file * F, struct request * R,
 	}
 
 	/* Write to the log file. */
-	if (logging_printf(F, "|%s|%s|%d|%s|%ld|%zu|%f",
+	if (logging_printf(F, "|%s|%s|%d|%s|%ld|%zu|%f|%s",
 	    R->op, R->logstr ? R->logstr : "",
-	    status, addr, t_micros, bodylen, capacity))
+	    status, addr, t_micros, bodylen, capacity,
+	    ddberr ? ddberr : ""))
```

Comment:
If there's an error and we don't have `ddberr`, the log line will end with a `|`.  Are you sure that you don't want to print `null` or `unknown_error` or something like that?

I'm also thinking about potentially parsing such a log-file.  I imagine that most parsing tools would be fine with a 0-width value between `|\n`, but if it was something like `|null\n` then there would be no doubt.

Reply:
Most lines will end with `|` since most lines will not have an error.  That's just fine.